polaris-core/src/main/java/org/apache/polaris/core/entity/PrincipalRoleEntity.java

```diff
@@ -65,6 +65,13 @@ public Builder(PrincipalRoleEntity original) {
       super(original);
     }
 
+    public Builder setFederated(Boolean isFederated) {
+      if (isFederated != null && isFederated) {
```

Comment:
This logic doesn't look right to me. This entity would come out federated:
```
new PolarisEntity.Builder()
. . .
.setFederated(true)
.setFederated(false)
.build()
```

Reply:
Yeah, the idea is to avoid adding any entry for non-federated identities. I don't think the logic snippet you posted is reasonable or should be supported. I'd be in favor of an IllegalStateException if someone typed that code. Happy to add that.

Reply:
Why should it not be supported? It sounds like you're saying that `.setFederated(false)` is never a valid call, in which case `setFederated` shouldn't take a boolean arg. Perhaps you are even looking for something like `buildFederated`?

Reply:
I support @eric-maynard 's point here. Current code is very risky in terms of latent bug, IMHO. A caller of `.setFederated(false)` can reasonably expect the property to become `false` regardless of any previous `set*` calls (which might have been made in a different context).

Reply:
@collado-mike : this is the only substantial issue with the current state if this PR from my POV.

The other topics got clarified in ML and doc comments, I think the PR is good to merge after we resolve the setter issue.

Reply:
I still feel it is a programming error for a builder to be invoked with this field twice. Is anyone opposed to an `IllegalStateException` in that case?

Reply:
From my POV it is way easier to allow multiple set calls with the latest call's value winning, than reducing flexibility by throwing `IllegalStateException`

Reply:
TBH, I don't see a reason, beyond building from an existing entity, to support setting a value in a builder twice. Modifying an existing entity is one reason to allow flexibility, but a builder has no reason for such flexibility. But, it's a minor point and I'm fine to concede if it allows us to move things along.